src/v/kafka/server/handlers/init_producer_id.cc

```diff
@@ -76,12 +76,27 @@ ss::future<response_ptr> init_producer_id_handler::handle(
               });
         }
 
-        if (!ctx.authorized(
-              security::acl_operation::idempotent_write,
-              security::default_cluster_name)) {
-            init_producer_id_response reply;
-            reply.data.error_code = error_code::cluster_authorization_failed;
-            return ctx.respond(reply);
+        bool permitted = false;
+        auto topics = ctx.metadata_cache().all_topics_metadata();
+        for (auto topic : topics) {
```

Comment:
I think you can replace these two lines with:

```
auto topics = ctx.metadata_cache().all_topics();
for (auto& topic : topics) {
```

to reduce the amount of work. `all_topics` avoids filling in all the partition information, and `auto&` avoids a memory copies.